gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/DatasetInfo.java

```diff
@@ -0,0 +1,409 @@
+          return datasetInfo.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -6615133444520365839L;
+
+  private final DatasetId datasetId;
+  private final List<Acl> acl;
+  private final Long creationTime;
+  private final Long defaultTableLifetime;
+  private final String description;
+  private final String etag;
+  private final String friendlyName;
+  private final String id;
+  private final Long lastModified;
+  private final String location;
+  private final String selfLink;
+
+  public static final class Builder {
+
+    private DatasetId datasetId;
+    private List<Acl> acl;
+    private Long creationTime;
+    private Long defaultTableLifetime;
+    private String description;
+    private String etag;
+    private String friendlyName;
+    private String id;
+    private Long lastModified;
+    private String location;
+    private String selfLink;
+
+    private Builder() {}
+
+    private Builder(DatasetInfo datasetInfo) {
+      this.datasetId = datasetInfo.datasetId;
+      this.acl = datasetInfo.acl;
+      this.creationTime = datasetInfo.creationTime;
+      this.defaultTableLifetime = datasetInfo.defaultTableLifetime;
+      this.description = datasetInfo.description;
+      this.etag = datasetInfo.etag;
+      this.friendlyName = datasetInfo.friendlyName;
+      this.id = datasetInfo.id;
+      this.lastModified = datasetInfo.lastModified;
+      this.location = datasetInfo.location;
+      this.selfLink = datasetInfo.selfLink;
+    }
+
+    /**
+     * Sets the dataset identity.
+     */
+    public Builder datasetId(DatasetId datasetId) {
+      this.datasetId = checkNotNull(datasetId);
+      return this;
+    }
+
+    /**
+     * Sets the dataset's access control configuration.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/access-control">Access Control</a>
+     */
+    public Builder acl(List<Acl> acl) {
+      this.acl = acl != null ? ImmutableList.copyOf(acl) : null;
+      return this;
+    }
+
+    Builder creationTime(Long creationTime) {
+      this.creationTime = creationTime;
+      return this;
+    }
+
+    /**
+     * Sets the default lifetime of all tables in the dataset, in milliseconds. The minimum value is
+     * 3600000 milliseconds (one hour). Once this property is set, all newly-created tables in the
+     * dataset will have an expirationTime property set to the creation time plus the value in this
+     * property, and changing the value will only affect new tables, not existing ones. When the
+     * expirationTime for a given table is reached, that table will be deleted automatically. If a
+     * table's expirationTime is modified or removed before the table expires, or if you provide an
+     * explicit expirationTime when creating a table, that value takes precedence over the default
+     * expiration time indicated by this property.
+     */
+    public Builder defaultTableLifetime(Long defaultTableLifetime) {
```

Comment:
should we say that this property is experimental and subject to change or removed?